File: pm_core_1/src/infrastructure/store/mapper/priority_scheme_item_mapper.go
package mapper

import (
	"github.com/serp/pm-core/src/core/domain/entity"
	"github.com/serp/pm-core/src/infrastructure/store/model"
)

type PrioritySchemeItemMapper struct{}

func NewPrioritySchemeItemMapper() *PrioritySchemeItemMapper {
	return &PrioritySchemeItemMapper{}
}

func (m *PrioritySchemeItemMapper) ToEntity(mdl *model.PrioritySchemeItemModel) *entity.PrioritySchemeItemEntity {
	if mdl == nil {
		return nil
	}
	return &entity.PrioritySchemeItemEntity{
		BaseEntity: entity.BaseEntity{
			ID:        mdl.ID,
			CreatedAt: mdl.CreatedAt.UnixMilli(),
			CreatedBy: mdl.CreatedBy,
			UpdatedAt: mdl.UpdatedAt.UnixMilli(),
			UpdatedBy: mdl.UpdatedBy,
		},
		TenantID:   mdl.TenantID,
		SchemeID:   mdl.SchemeID,
		PriorityID: mdl.PriorityID,
		Sequence:   mdl.Sequence,
	}
}

func (m *PrioritySchemeItemMapper) ToModel(e *entity.PrioritySchemeItemEntity) *model.PrioritySchemeItemModel {
	if e == nil {
		return nil
	}
	return &model.PrioritySchemeItemModel{
		BaseModel: model.BaseModel{
			ID:        e.ID,
			CreatedBy: e.CreatedBy,
			UpdatedBy: e.UpdatedBy,
		},
		TenantID:   e.TenantID,
		SchemeID:   e.SchemeID,
		PriorityID: e.PriorityID,
		Sequence:   e.Sequence,
	}
}

func (m *PrioritySchemeItemMapper) ToEntities(models []*model.PrioritySchemeItemModel) []*entity.PrioritySchemeItemEntity {
	if models == nil {
		return nil
	}
	entities := make([]*entity.PrioritySchemeItemEntity, 0, len(models))
	for _, mdl := range models {
		if e := m.ToEntity(mdl); e != nil {
			entities = append(entities, e)
		}
	}
	return entities
}

func (m *PrioritySchemeItemMapper) ToModels(entities []*entity.PrioritySchemeItemEntity) []*model.PrioritySchemeItemModel {
	if entities == nil {
		return nil
	}
	models := make([]*model.PrioritySchemeItemModel, 0, len(entities))
	for _, e := range entities {
		if mdl := m.ToModel(e); mdl != nil {
			models = append(models, mdl)
		}
	}
	return models
}

File: pm_core/src/main/java/serp/project/pmcore/infrastructure/store/mapper/PrioritySchemeItemMapper.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.infrastructure.store.mapper;

import org.springframework.stereotype.Component;
import serp.project.pmcore.core.domain.entity.PrioritySchemeItemEntity;
import serp.project.pmcore.infrastructure.store.model.PrioritySchemeItemModel;

import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

@Component
public class PrioritySchemeItemMapper extends BaseMapper {

    public PrioritySchemeItemEntity toEntity(PrioritySchemeItemModel model) {
        if (model == null) { return null; }
        return PrioritySchemeItemEntity.builder()
                .id(model.getId())
                .tenantId(model.getTenantId())
                .schemeId(model.getSchemeId())
                .priorityId(model.getPriorityId())
                .sequence(model.getSequence())
                .createdAt(localDateTimeToLong(model.getCreatedAt()))
                .createdBy(model.getCreatedBy())
                .updatedAt(localDateTimeToLong(model.getUpdatedAt()))
                .updatedBy(model.getUpdatedBy())
                .build();
    }

    public PrioritySchemeItemModel toModel(PrioritySchemeItemEntity entity) {
        if (entity == null) { return null; }
        return PrioritySchemeItemModel.builder()
                .id(entity.getId())
                .tenantId(entity.getTenantId())
                .schemeId(entity.getSchemeId())
                .priorityId(entity.getPriorityId())
                .sequence(entity.getSequence())
                .createdAt(longToLocalDateTime(entity.getCreatedAt()))
                .createdBy(entity.getCreatedBy())
                .updatedAt(longToLocalDateTime(entity.getUpdatedAt()))
                .updatedBy(entity.getUpdatedBy())
                .build();
    }

    public List<PrioritySchemeItemEntity> toEntities(List<PrioritySchemeItemModel> models) {
        if (models == null) { return Collections.emptyList(); }
        return models.stream().map(this::toEntity).collect(Collectors.toList());
    }

    public List<PrioritySchemeItemModel> toModels(List<PrioritySchemeItemEntity> entities) {
        if (entities == null) { return Collections.emptyList(); }
        return entities.stream().map(this::toModel).collect(Collectors.toList());
    }
}
